src/material-experimental/mdc-chips/chip-row.spec.ts

```diff
@@ -226,6 +235,130 @@ describe('MDC-based Row Chips', () => {
         });
       });
     });
+
+    describe('editable behavior', () => {
+      beforeEach(() => {
+        testComponent.editable = true;
+        fixture.detectChanges();
+      });
+
+      it('should apply the mdc-chip--editable class', () => {
+        expect(chipNativeElement.classList).toContain('mdc-chip--editable');
+      });
+
+      it('should begin editing on double click', () => {
+        chipInstance._dblclick(createMouseEvent('dblclick'));
+        expect(chipNativeElement.classList).toContain('mdc-chip--editing');
+      });
+
+      it('should begin editing on ENTER', () => {
+        chipInstance.focus();
+        const primaryActionElement = chipNativeElement.querySelector('.mdc-chip__primary-action')!;
+        chipInstance._keydown(createKeyboardEvent('keydown', ENTER, 'Enter', primaryActionElement));
+        expect(chipNativeElement.classList).toContain('mdc-chip--editing');
+      });
+    });
+
+    describe('editing behavior', () => {
+      let editInputInstance: MatChipEditInput;
+      let chipContentElement: HTMLElement;
+
+      beforeEach(() => {
+        testComponent.editable = true;
+        fixture.detectChanges();
+        chipInstance._dblclick(createMouseEvent('dblclick'));
+        spyOn(testComponent, 'chipEdit');
+        fixture.detectChanges();
+
+        const editInputDebugElement = fixture.debugElement.query(By.directive(MatChipEditInput))!;
+        editInputInstance = editInputDebugElement.injector.get<MatChipEditInput>(MatChipEditInput);
+
+        chipContentElement =
+          chipNativeElement.querySelector('.mat-chip-row-focusable-text-content') as HTMLElement;
+      });
+
+      function keyDownOnPrimaryAction(keyCode: number, key: string) {
+        const primaryActionElement = chipNativeElement.querySelector('.mdc-chip__primary-action')!;
+        chipInstance._keydown(createKeyboardEvent('keydown', keyCode, key, primaryActionElement));
```

Comment:
I think we have some helper functions for typing in elements: https://github.com/angular/components/blob/master/src/cdk/testing/testbed/fake-events/type-in-element.ts (they need to be imported from `@angular/cdk/testing/private` though, they're re-exported there for our unit tests)

Reply:
I wasn't able to use this since it triggered the input event instead of keydown, but I switched to using the dispatchEvent helper instead of calling a private function.